app/(main)/o/[slug]/conversations/[id]/summary.tsx

```diff
@@ -660,7 +665,53 @@ export default function Summary({ className, source, slug, onCloseClick = () =>
           <Image src={getRagieStreamPath(slug, source.imageUrl)} alt="Image" width={500} height={500} />
         </div>
       )}
-      <div className="text-[12px] font-bold mb-4">Summary</div>
+
+      {((source.mergedRanges && source.mergedRanges.length > 0) || (source.startPage && source.endPage)) && (
```

Comment:
This is getting a little complicated, might be worth splitting out into its own component.